charts/pega/charts/constellation/templates/clln-deployment.yaml

```diff
@@ -0,0 +1,36 @@
+#Deploy only when the constellation flag has been enabled in the values yaml.
+{{ if and .Values.enabled (eq .Values.enabled true) }}
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: constellation
+  labels:
+    app: constellation
+    # component: constellation
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: constellation
+      # component: constellation
+  template:
+    metadata:
+      labels:
+        app: constellation
+        # component: constellation
+    spec:
+      imagePullSecrets:
+      - name: regcred
+      containers:
+      # cli - the sidecar - here be the brains
+      - name: clln-cli
+        #imagePullPolicy: Never
+        image: {{ .Values.dockerRepos }}
+        args:
+        - port=3000
+        - infinity={{ .Values.infinityUrl }}
```

Comment:
This should use the appropriate service from the parent chart.  No URL should be necessary.

Reply:
@dcasavant Yes ok please tell us how to specify it.

Reply:
Discussed with @kishorv10 & @WhiteWalkers, As suggested by them. We can loop over tiers and pull out domain parameter of nodeType "WebUser". Constellation need this infinity to make call from client/browser(REST) to pega-infinity, which wont be resolved by service name. Hence required url not only service. 
Additionally, if its onprem we can loop over. And if this value is set in charts/pega/charts/constellation/values.yaml, then we will directly use this. 
Please let us know your comment @dcasavant

Reply:
I don't follow - if the client has accessed a URL, then that URL is already available from the client side.  The Constellation service would not need to know if the URL is only being accessed from the client, right?

Reply:
From what I can tell, this url is used from within a constellation instance to access a pega instance.  Wouldn't these be pods within the same kubernetes cluster?  If so, you should just be able to pass in the pega-web service URL (e.g. http://pega-web) or construct the URL via the pega-web service environment variables (e.g. PEGA_WEB_PORT_80_TCP_ADDR)

Reply:
As per our discussion earlier today, I believe the suggestion to use http://pega-web/prweb as the infinity URL used by constellation should work for this use case.  There should be no need to expose this as a deployment parameter in the values.yaml associated with the deployment.

Reply:
To clarify, the service name (and therefore the URL to access the service) is based on the value provided as the tier name in the values.yaml, so it should be derived from the tier name specified (e.g. The service associated with tier name, "blah" will be accessible within the kubernetes cluster via http://pega-blah).